app/src/main/java/com/example/mvvmsampleapp/Main/MainActivity.kt

```diff
@@ -0,0 +1,123 @@
+package com.example.mvvmsampleapp.Main
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.activity.viewModels
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.mvvmsampleapp.R
+import com.example.mvvmsampleapp.adapters.MainAdapter
+import com.example.mvvmsampleapp.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private val mainViewModel: MainViewModel by viewModels()
+    private val mainAdapter = MainAdapter()
+    private var bundle: Bundle?=null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (savedInstanceState != null) {
+            bundle=savedInstanceState
+        }
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setupUI()
+        fetchUsersBtn()
+        setupLoadingObserver()
+        setupUsersObserver()
+        setupEmptyObserver()
+        setupFailureObserver()
+    }
+
+    fun fetchUsersBtn() {
+        if(bundle==null) {
+            mainViewModel.onFetchUsersBtnClicked()}
+        else Log.d("youssef2","youssef25 $bundle")
+        //binding.btnGetUsers.setOnClickListener { mainViewModel.onFetchUsersBtnClicked() }
+    }
+
+    private fun setupUI() {
+        binding.recyclerView.layoutManager = LinearLayoutManager(this)
+        binding.recyclerView.addItemDecoration(
+            DividerItemDecoration(
+                binding.recyclerView.context,
+                (binding.recyclerView.layoutManager as LinearLayoutManager).orientation
+            )
+        )
```

Comment:
It is good but we can abbreviation it by add this line line in `            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
` in recyclerView xml